### PR TITLE
fix!: for schemas of data processors

### PR DESCRIPTION
**Pull request by anaprietonem**
## Description
Pydantic schemas not resolving correcting and getting into the model's metadata and internal class names. Resolve this by:
- Moving data-processors schemas to models. Initially left in training since those are defined in the 'data' config. However those are ultimately model layers - note this change is a **breaking change**
- Test at the beginning of the training that there is no pydantic schemas in the model, otherwise crash
- Add test in the data processors to ensure all schemas work ( This is to mitigate the fact that default configs just include test for normalizers, so we do not test others like imputers or post-processors commonly used for Earth System Models)

## What problem does this change solve?
Issue - https://github.com/ecmwf/anemoi-core/issues/421

##  Additional notes ##

***As a contributor to the Anemoi framework, please ensure that your changes include unit tests, updates to any affected dependencies and documentation, and have been tested in a parallel setting  (i.e., with multiple GPUs). As a reviewer, you are also responsible for verifying these aspects and requesting changes if they are not adequately addressed. For guidelines about those please refer to https://anemoi.readthedocs.io/en/latest/***

By opening this pull request, I affirm that all authors agree to the [Contributor License Agreement.](https://github.com/ecmwf/codex/blob/main/Legal/contributor_license_agreement.md)

### CODE REVIEW

**Comment by anaprietonem** [comment]
@sahahner I have extended the tests, could you check these works fine and have a look?
@gmertes  I have now added a check in our ModelCheckpoint callback to catch this problem. Tested it in main and caught the problem fine. Should just be done at the beginning of the training.

**Review by sahahner — APPROVED** [review]
Thank you for correcting the schemas for the pre/postprocessors. They work fine for my config files, and the tests for the schemas look good to me.

File: training/src/anemoi/training/utils/checkpoint.py [hunk 1]
```diff
@@ -113,3 +115,21 @@ def freeze_submodule_by_name(module: nn.Module, target_name: str) -> None:
         else:
             # Recursively search within children
             freeze_submodule_by_name(child, target_name)
+
+
+class LoggingUnpickler(pickle.Unpickler):
+    def find_class(self, module: str, name: str) -> str:
+        if "anemoi.training.schemas" in module:
```
Comment (gmertes) [inline training/src/anemoi/training/utils/checkpoint.py]: Should we also check for any `anemoi.training` reference just to be sure?

Comment (anaprietonem) [reply training/src/anemoi/training/utils/checkpoint.py]: Yes I wondered, I guess it should be fair to enforce that since that's checking the 'plain torch' model. Will update it

**Comment by anaprietonem** [comment]
@gmertes I have update the check to do just 'anemoi.training' rather the schemas specifically. Updated the branch and tested it and seems fine (ie if I test this with the current main it crashes as expected).

**Review by gmertes — APPROVED** [review]
Very nice, thanks for fixing this bug!
